### PR TITLE
Fix lexing of large integer literals

### LINKED ISSUE #16: Large integer literals in Lexer
The value of an integer should not be restricted in the Lexer, therefore any integer (that is reasonable to store in the memory, ~1000 characters at least) should be accepted by the Lexer.

Current behaviour: `error: [...] integer literal too large`

### PR DESCRIPTION
**Pull request by g4rgoil**
Resolves #16

### CODE REVIEW

**Review by larsk21 — COMMENTED** [review]
Please change the corresponding case in `Token::getStringRepresentation` where we have a literal value instead of an int value now.

File: src/main/java/edu/kit/compiler/data/Literal.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package edu.kit.compiler.data;
+
+import lombok.EqualsAndHashCode;
+
+/**
+ * Represents a literal value in its textual form.
+ */
+@EqualsAndHashCode(callSuper = false)
+public final class Literal {
```
[the remark this answers is not in the document]
Comment (g4rgoil) [reply src/main/java/edu/kit/compiler/data/Literal.java:9]: For us it's only going to be integer literals. In theory the class could be extended to for example support floating point literals.

File: src/main/java/edu/kit/compiler/data/Literal.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        return value.toString();
+    }
+
+    /**
+     * Returns the value of the literal when parsed as an integer.
+     * 
+     * @return the integer value represented by the literal.
+     * @throws NumberFormatException if the literal does not represent a valid integer.
+     */
+    public int intValue() {
```
[the remark this answers is not in the document]
Comment (g4rgoil) [reply src/main/java/edu/kit/compiler/data/Literal.java:32]: I can definitely do that, though I'd probably implement it using a try-catch internally.

File: src/main/java/edu/kit/compiler/data/Literal.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+     * 
+     * @param value the literal's textual representation.
+     */
+    public Literal(String value) {
+        this.value = value;
+    }
+
+    @Override
+    public String toString() {
+        return value.toString();
```
[the remark this answers is not in the document]
Comment (g4rgoil) [reply src/main/java/edu/kit/compiler/data/Literal.java:23]: We need a method to get the string representation of the literal. This is needed in the `getStringRepresentation` of `Token`.

[the remark this answers is not in the document]
Comment (larsk21) [reply src/main/java/edu/kit/compiler/data/Literal.java:23]: Yes, but `value` is already a string, isn't it?

File: src/main/java/edu/kit/compiler/data/Literal.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (larsk21) [reply src/main/java/edu/kit/compiler/data/Literal.java:32]: I was thinking about avoiding the overhead of throwing an Exception, though I don't know if that's possible here.

File: src/main/java/edu/kit/compiler/data/Literal.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (larsk21) [reply src/main/java/edu/kit/compiler/data/Literal.java:9]: Ah, I see.

File: src/main/java/edu/kit/compiler/data/Literal.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (g4rgoil) [reply src/main/java/edu/kit/compiler/data/Literal.java:23]: Yes, I'm just using `toString` as a convenient interface.

File: src/main/java/edu/kit/compiler/data/Literal.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (g4rgoil) [reply src/main/java/edu/kit/compiler/data/Literal.java:32]: Using exceptions for this is not ideal. Unfortunately I don't know of a method like `isValidInteger` or anything comparable in the standard library.

File: src/main/java/edu/kit/compiler/data/Literal.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (larsk21) [reply src/main/java/edu/kit/compiler/data/Literal.java:23]: Are we talking about the same thing? :D Couldn't you just write `return value;`?

[the remark this answers is not in the document]
Comment (g4rgoil) [reply src/main/java/edu/kit/compiler/data/Literal.java:23]: Yeah sorry, my bad. You're right, I don't need the `toString()` here.

**Comment by g4rgoil** [comment]
> Please change the corresponding case in `Token::getStringRepresentation` where we have a literal value instead of an int value now.

`Token::getStringRepresentation()` is fixed, though I would like to change the entire switch case to a switch expression in order to avoid the default case.

**Review by larsk21 — APPROVED** [review]

**Pull request merged**